src/data_access/InMemoryGameDataAccessObject.java

```diff
@@ -19,4 +21,58 @@ public Game getGameByID(String gameID) {
     public void save(Game game) {
         games.put(game.getID(), game);
     }
+
+    @Override
+    public HashMap<String, Object> avgStats() {
```

Comment:
I'm not a huge fan of returning arbitrary maps of objects since there is no guarantee of what they hold and the only way of using them properly is by having existing knowledge of what keys are supposed to be in the map. This is not very maintainable in the long run, especially if we create more game DAO's (which we will). Since we know exactly what data we want to store and access, I think we should create some sort of `LifetimeStatistics` entity (or whatever appropriate name you think) and return that instead. I think this adheres to CA better as well.

This doesn't have to be super complicated either, as it could look something like this:

```java
// LifetimeStatistics.java
public interface LifetimeStatistics {
    int getAverageScore();
    int getAverageInitialLives();
    String getTopDifficulty();
    String getTopGenre();
}

// CommonLifetimeStatistics.java
public class CommonLifetimeStatistics implements LifetimeStatistics  {

    // instance variables
    // ...

    public CommonLifetimeStatistics(int averageScore, int averageInitialLives, String topDifficulty, String topGenre) {
        // set instance variables
    }

    // interface methods (getters for instance variables)
    // ...
}
```
Then replace everywhere that uses `HashMap<String, Object>` to use `LifetimeStatistics` and change the map.get calls to use the corresponding getter method.